src/computer_player.c

```diff
@@ -0,0 +1,604 @@
+    }
+    else if (Y_count > R_count && Y_count > B_count && Y_count > G_count)
+    {
+        return YELLOW;
+    }
+
+    return 0;
+}
+
+/* function that choose the right card to draw */
+/*
+* @brief: function that choose the right card to draw
+*          if player card # > 1
+*           case 1 : no matched color and no matched number
+*             - if have wild card, draw wild if wild card > with color with the most occurence
+*             - if have skip, use skip
+*             - if not have wild or skip, take card from pile
+*            case 2 : have the matched color and no matched number
+*             - choose the largest number on that color
+*             - if matched color > 3, use match draw/draw2/skip card based on the color with the most occurence
+*
+*            case 3 : no matched color, but have matched number
+*             - if have multiple, choose the color with the most occurence and corresponding action card
+*             - if have only one, get rid of it
+*
+*            case 4 : have matched color and match number
+*             - quickly get rid of matched numbers in the hand first.
+*             - if occurence of
+*
+*         if player card # = 1
+*            - use action card if possibile
+*            - avoid play the last color that the player played
+*
+* @return int the case number
+*/
+
+
+int pick_case(struct CARD card, struct DECK *hand_card)
+{
+    struct DECK *ptr1 = hand_card, *ptr2 = hand_card;
+    int commonColor = 0, commonName = 0;
+    while (ptr1 != NULL) {
+        if (card.color == ptr1->card.color) {
+            commonColor = 1;
+        }
+        ptr1 = ptr1->next;
+    }
+    while (ptr2 != NULL) {
+        if (card.name == ptr2->card.name) {
+            commonName = 1;
+        }
+        ptr2 = ptr2->next;
+    }
+    if (commonColor && commonName) {
+        return 4;
+    }
+    else if (!commonColor && commonName) {
+        return 3;
+    }
+    else if (commonColor && !commonName) {
+        return 2;
+    }
+    else if (!commonColor && !commonName) {
+        return 1;
+    }
+    return 0;
+}
+
+/*
+* @brief: function that play card from handcard list
+*
+* @return void
+*/
+void play_card(struct CARD card, struct DECK **head)
+{
+    struct DECK *prev = *head;
+    struct DECK *temp = *head;
+    // head node
+    if (temp != NULL && temp->card.name == card.name && temp->card.color == card.color)
+    {
+        *head = temp->next;
+        free(temp);
+        return;
+    }
+
+    while (temp != NULL && temp->card.name == card.name && temp->card.color == card.color)
+    {
+        prev = temp;
+        temp = temp->next;
+    }
+    // if not present
+    if (temp == NULL)
+        return;
+
+    // delete
+    prev->next = temp->next;
+
+    free(temp);
+}
+/*
+* @brief: function that check if a card exist in the handcard deck
+*
+* @return 1 exist, 0 not exist
+*/
+
+int ifexists(struct DECK **handcard, enum CardName inputCardName, enum CardColor inputCardColor)
```

Comment:
enum CardName inputCardName, enum CardColor inputCardColor should be replaced as  const Card_t input_card.

Reply:
Can not do that, this function aims to find whether if a specific card exists in the hand card list. If the input is type CARD, I will need to define a card first and then check. it is more convenient this way :)